pdns/dnsdist-lua.cc

```diff
@@ -1887,7 +1889,7 @@ static void setupLuaConfig(LuaContext& luaCtx, bool client, bool configCheck)
       }
     }
 
-    closedir(dirp);
+    dirp.reset();
```

Comment:
I'm fine with the call to `reset()` here, but I have a slight preference for adding a dummy scope instead, or separating this part into a function altogether.